pkg/codegen/nodejs/utilities.go

```diff
@@ -99,6 +99,9 @@ func makeValidIdentifier(name string) string {
 		}
 	}
 	name = builder.String()
+	if strings.HasPrefix(name, "pulumi_") {
+		name = strings.TrimPrefix(name, "pulumi_")
+	}
```

Comment:
Putting this here in `makeValidIdentifier` seems like it would affect more than just imports. It looks like the package name is inconsistently handled. I'd look at `genPreamble` to figure out why the import is generated as `pulumi_eventstorecloud`. 

The schema for Pulumi package says its name is `eventstorecloud`: https://github.com/EventStore/pulumi-eventstorecloud/blob/main/provider/cmd/pulumi-resource-eventstorecloud/schema.json

But the NPM package is `@eventstore/pulumi-eventstorecloud`. 

Is `genPreamble` generating the `import * as` name from the name of the NPM package instead of the name of the Pulumi package? That'd be my hunch.

Reply:
The package name is correctly `@eventstore/pulumi-eventstorecloud`. We strip it to `pulumi-eventstorecloud` with [Base()](https://github.com/pulumi/pulumi/blob/master/pkg/codegen/nodejs/gen_program.go#L322), then swap the `-` for `_`. It looks like we'll need a way of determining this provider originates from a non-pulumi repo, similar to the change [here](https://github.com/pulumi/pulumi/pull/10485)

Reply:
Right - I think the issue is that we're working with the wrong string here. The issue isn't detecting whether it's "Pulumi owned", it's that we're doing:

```
makeValidIdentifier(NPM package name after the /)
```

Instead of:

```
makeValidIdentifier(Pulumi package name)
```

Where:

* NPM package name: `@eventstore/pulumi-eventstorecloud`
* Pulumi package name: `eventstorecloud`

The NPM package name ending in the same string as the Pulumi package name here is a coincidence and a misdirect.